design/FY2017/NFP-Enhancement for External Shading Calculation .md

```diff
@@ -0,0 +1,104 @@
+
+**Revised: June 27, 2017**
+
+# Justification for New Feature 
+
+External shading, i.e., shading of an exterior building surface, is closely connected with energy use in buildings for heating, cooling, and lighting and with the occupants’ visual and thermal comfort. Accurate calculation of the shading that building environment, building elements or shading devices may cast on the building envelope, plays a key role in thermal load calculations and building energy simulation. However, external shading calculation for a target building in an urban context can be considerably slow considering many surrounding surfaces may shade a target surface. Various external tools can be used to more efficiently pre-calculate the shading fractions for many target buildings in the urban context. We propose to add links to EnergyPlus to enable the import of external shading calculation results, and saving the EnergyPlus shading calculation results for reuse.  
+
+# Team Discussion so far
+Michael Witte and Stuart Mentzer provided suggestions on input details and external tool resources.
+
+# Overview 
+Solar shading affects energy use in a building by reducing the solar heat gains received by opaque surfaces or the transmitted and absorbed solar radiation of windows. External shading also influences daylighting level in a room and the view to the exterior. Accurate calculation of solar shading on building exterior surfaces is of great importance in whole building energy simulation. Modeling a building in the urban context may involve many shading surfaces from adjacent buildings for a target building, which can significantly slow down EnergyPlus simulations due to the currently implemented shading calculation algorithms. Furthermore, for urban building energy simulation which involves many buildings, there can be more efficient ways to do the shading calculations than the current way of doing shading calculations for every building separately in EnergyPlus. There is also the potential of using GPU and other parallel computing techniques for building shading calculations.
+
+Various simulation tools, such as Radiance and Daysim, employing state-of-the-art ray-tracing simulation techniques, can be used to pre-calculate the shading fractions for each exterior building surface. However, in the current implementation, EnergyPlus doesn’t allow overriding exterior shading calculations with external data either by EMS or co-simulation interface. Considering a large amount of information and the complexity of calculating the dynamic shading in EnergyPlus, allowing this external input would enable a huge potential to speed up EnergyPlus in urban-scale simulations. Another benefit is to enable reusing the shading results for parametric runs which usually do not change external shading. Adding this feature to EnergyPlus would also allow OpenStudio to run radiance simulations before the EnergyPlus simulation and reusing the radiance calculations for shading. 
```

Comment:
And the same building location and orientation. Would need to add a checksum if we want to detect cached shading calcs automatically. And maybe also if we want to do this manually, just to avoid surprising results.